trace_extra/memory_interceptor.hh

```diff
@@ -0,0 +1,49 @@
+//
+// Created by mc2262 on 15/07/22.
+//
+
+#pragma once
+#include <perfetto.h>
+#include <iostream>
+#include <fstream>
+#include "dynamorio/clients/drcachesim/common/trace_entry.h"
+
+using namespace std;
+
+// MemOp definition from QEMULogEntryMem in qemu_log_entry.proto
+  typedef enum {
+    LOAD = 0,
+    STORE = 1,
+    CLOAD = 2,
+    CSTORE = 3,
+  } MemOp;
+
+class MyInterceptor : public perfetto::Interceptor<MyInterceptor> {
```

Comment:
Maybe rename the interceptor to something more meaningful, say `DynamorioTraceInterceptor` or something like that.